packages/react-next/src/components/Modal/Modal.base.tsx

```diff
@@ -183,12 +90,30 @@ export class ModalBase extends React.Component<IModalProps, IDialogState> implem
       onLayerDidMount,
       isModeless,
       dragOptions,
-    } = this.props;
-    const { isOpen, isVisible, hasBeenOpened, modalRectangleTop, x, y, isInKeyboardMoveMode } = this.state;
+      onDismissed,
+    } = props;
 
-    if (!isOpen) {
-      return null;
-    }
+    const internalState = useConst<IModalInternalState>(() => ({
+      onModalCloseTimer: 0,
+      allowTouchBodyScroll: false,
+      hasRegisteredKeyUp: false,
+      scrollableContent: null,
+      lastSetXCoordinate: 0,
+      lastSetYCoordinate: 0,
+      events: new EventGroup({}),
+    }));
+
+    const [isModalOpen, { setFalse: setModalClose, setTrue: setModalOpen }] = useBoolean(isOpen);
+
+    const [isModalMenuOpen, { toggle: toggleModalMenuOpen, setFalse: setModalMenuClose }] = useBoolean(false);
```

Comment:
Arguably this should also just use `useState`. Toggling can be handled without direct access to the current value by doing `setIsModalMenuOpen(prevValue => !prevValue)`. Though since you're using the toggle callback frequently I could go either way on this one.